### PR TITLE
drivers/spi: stm32: Move pin configuration to device tree

### PR DESCRIPTION
**Pull request by erwango**
This PR introduces SPI pin control configuration using device tree on STM32 driver

### CODE REVIEW

File: boards/arm/96b_carbon/96b_carbon.dts [hunk 1]
```diff
@@ -86,6 +86,10 @@
 &spi1 {
 	status = "okay";
 
+	pinctrl-0 = <&spi1_miso_pa6 &spi1_mosi_pa7
+		     &spi1_sck_pa5 &spi1_nss_pa4>;
+	pinctrl-names = "default";
```
Comment (galak) [inline boards/arm/96b_carbon/96b_carbon.dts:91]: do we want to keep `pinctrl-names` ?

**Review by tbursztyka — APPROVED** [review]

File: boards/arm/96b_carbon/96b_carbon.dts (hunk 1 above)
Comment (erwango) [reply boards/arm/96b_carbon/96b_carbon.dts]: Correct, fixed!

**Review by gmarull — APPROVED** [review]

**Review by ABOSTM — APPROVED** [review]
LGTM

File: boards/arm/disco_l475_iot1/disco_l475_iot1.dts [hunk 2]
```diff
@@ -121,9 +121,13 @@
 	status = "okay";
 };
 
+&spi3_miso_pc11{ slew-rate = "very-high-speed";};
```
Comment (martinjaeger) [inline boards/arm/disco_l475_iot1/disco_l475_iot1.dts:124]:
```suggestion
&spi3_miso_pc11 { slew-rate = "very-high-speed"; };
```

Nits comment to align style with #29267

Comment (erwango) [reply boards/arm/disco_l475_iot1/disco_l475_iot1.dts]: Txs, fixed

**Review by martinjaeger — APPROVED** [review]

**Pull request merged**